static/src/js/zustand/selectors/query.ts

```diff
@@ -35,14 +35,50 @@ import {
 import { prepareSubscriptionQuery, removeDisabledFieldsFromQuery } from '../../util/subscriptions'
 
 /**
- * Retrieve current collection query information
+ * Retrieve current collection query information for collection searches
  */
 export const getCollectionsQuery = (state: EdscStore) => state.query?.collection || {}
 
+/**
+ * Retrieve NLP collection data
+ */
+export const getNlpCollection = (state: EdscStore) => state.query?.nlpCollection || null
+
+/**
+ * Retrieve NLP spatial data
+ */
+export const getNlpSpatialData = (state: EdscStore) => {
+  const nlpCollection = getNlpCollection(state)
+
+  return nlpCollection?.spatial || null
+}
+
+/**
+ * Retrieve NLP temporal data
+ */
+export const getNlpTemporalData = (state: EdscStore) => {
+  const nlpCollection = getNlpCollection(state)
+
+  return nlpCollection?.temporal || null
+}
+
 /**
  * Retrieve current collection spatial information
+ * Returns empty spatial arrays for NLP search (NLP doesn't use CMR spatial parameters), otherwise regular spatial data
```

Comment:
You're still reusing pieces of the application that don't make sense. A collections query is not the same as an NLP query. Stop trying to share functionality between the two things. 

Anywhere that is using the query information needs to be specific about what it is requesting so that the code is easy to understand.